src/main/java/net/sf/jabref/logic/integrity/BibtexkeyChecker.java

```diff
@@ -0,0 +1,33 @@
+package net.sf.jabref.logic.integrity;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import net.sf.jabref.logic.integrity.IntegrityCheck.Checker;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.model.entry.FieldName;
+
+public class BibtexkeyChecker implements Checker {
+
+    @Override
+    public List<IntegrityMessage> check(BibEntry entry) {
+        Optional<String> valuekey = entry.getField(BibEntry.KEY_FIELD);
+        Optional<String> valueauthor = entry.getField(FieldName.AUTHOR);
+        Optional<String> valuetitle = entry.getField(FieldName.TITLE);
+        Optional<String> valueyear = entry.getField(FieldName.YEAR);
+        String authortitleyear = entry.getAuthorTitleYear(100);
+
+        if (!valueauthor.isPresent() || !valuetitle.isPresent() || !valueyear.isPresent()) {
```

Comment:
Judging based on the code, I think getAuthorTitleYear would return `N/A: N/A (N/A)` in the case where the fields are missing. I would suggest that you change the order of the if statements.
If the key is empty, then we definitely want to report a message. But if then all the author, title and year are empty, we use BibEntry.toString (it is a bit ugly) or otherwise we use getAuthorTitleYear.